Source/FieldSolver/SpectralSolver/SpectralFieldDataRZ.H

```diff
@@ -73,9 +73,14 @@ class SpectralFieldDataRZ
          */
         void CopySpectralDataComp (const int src_comp, const int dest_comp)
         {
-            // The last two arguments represent the number of components and
-            // the number of ghost cells to perform this operation
-            Copy(fields, fields, src_comp, dest_comp, n_rz_azimuthal_modes, 0);
+            const int nmodes = n_rz_azimuthal_modes;
+            for (int imode=0 ; imode < nmodes ; imode++)
```

Comment:
I would add a comment that explains why the shift is needed, and refers to the allocation of `fields` in the constructor of the class. Also, I think we could use `n_rz_azimuthal_modes` directly in the `for` loop, instead of allocating an extra variable:
```suggestion
            // The fields of each mode are grouped together, so that the index of a 
            // field for a specific mode is given by field_index + mode*n_fields
            // (see allocation of fields in SpectralFieldDataRZ.cpp)
            for (int imode=0; imode < n_rz_azimuthal_modes; imode++)
```

I would do this here as well as in the other two functions below.

Reply:
Thank you for the review, @EZoni. This was fixed in 5d17ad222c3149eb5463fa0570ce46f205259e4f and 81530393add446fddd62909b3daf280bf3ed043c.